src/v2/guide/components-dynamic-async.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Dynamic & Async Components
+type: guide
+order: 105
+---
+
+> This page assumes you've already read the [Components Basics](components.html). Read that first if you are new to components.
+
+## `keep-alive` with Dynamic Components
+
+Earlier, we used the `is` attribute to switch between components in a tabbed interface:
+
+```html
+<component v-bind:is="currentTabComponent"></component>
+```
+
+When switching between these components though, you'll sometimes want to maintain their state or avoid re-rendering for performance reasons. In these cases, you can wrap a dynamic component with a `<keep-alive>` element:
+
+``` html
+<!-- Inactive components will be cached! -->
+<keep-alive>
+  <component v-bind:is="currentTabComponent"></component>
+</keep-alive>
```

Comment:
This, and the API don't fully show off the benefits of `keep-alive`, IMO, but that's out of scope of this PR, so I've filed an [issue for myself](https://github.com/vuejs/vuejs.org/issues/1493) to take care of it when this is merged.

Reply:
I agree and think that's a great idea!

Reply:
Couldn't help myself and added it. 😅 Your idea was too good!